tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs

```diff
@@ -147,6 +147,9 @@ private void OnBeginRequest(object sender, EventArgs eventArgs)
                 var security = Security.Instance;
                 if (security.Settings.Enabled)
                 {
+                    security.InstrumentationGateway.RaiseRequestStart(httpContext, httpContext.Request, scope.Span);
+                    security.InstrumentationGateway.RaiseBlockingOpportunity(httpContext, scope, tracer.Settings, (_) => { });
```

Comment:
are we calling `RaiseBlockingOpportunity`  only on request start? regarding the core version, it's called after every aspnet (and mvc events) where we get more info to pass to the waf (path params, body available, end request with new status codes...)

Reply:
fixed in b28b1151d

Reply:
fixed in b28b1151d

Reply:
so there's some security in [ `ControllerContextExtensions` ](https://github.com/DataDog/dd-trace-dotnet/blob/44723e53da9d1a3e3a5ab2b0df3bf8cec9486d4b/tracer/src/Datadog.Trace/AppSec/ControllerContextExtensions.Framework.cs#L45) and [here](https://github.com/DataDog/dd-trace-dotnet/blob/7a74b9888efb41be24f5e4f7bd98090575dbe29e/tracer/src/Datadog.Trace/AspNet/TracingHttpModule.cs#L161) and I'm not sure about some other places, but I guess we could add this later, as it's mostly to block ips, which will be normally available at request start